hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/safemode/TestSCMSafeModeManager.java

```diff
@@ -571,10 +701,18 @@ public void testSafeModePipelineExitRule() throws Exception {
           config, containers, null, pipelineManager, queue, serviceManager,
           scmContext);
 
-      queue.fireEvent(SCMEvents.NODE_REGISTRATION_CONT_REPORT,
-          HddsTestUtils.createNodeRegistrationContainerReport(containers));
+      SCMDatanodeProtocolServer.NodeRegistrationContainerReport nodeRegistrationContainerReport =
+          HddsTestUtils.createNodeRegistrationContainerReport(containers);
+      queue.fireEvent(SCMEvents.CONTAINER_REGISTRATION_REPORT, nodeRegistrationContainerReport);
+
       assertTrue(scmSafeModeManager.getInSafeMode());
 
+      for (DatanodeDetails datanodeDetail : pipeline.getNodes()) {
+        SCMDatanodeProtocolServer.NodeRegistrationContainerReport nodeRegistrationReport =
+            new SCMDatanodeProtocolServer.NodeRegistrationContainerReport(datanodeDetail, null);
+        queue.fireEvent(SCMEvents.NODE_REGISTRATION_CONT_REPORT, nodeRegistrationReport);
+      }
+
```

Comment:
This is not needed. Instead, we only need to fire `queue.fireEvent(SCMEvents.NODE_REGISTRATION_CONT_REPORT, nodeRegistrationContainerReport);` once, when we fire the `CONTAINER_REGISTRATION_REPORT` event.

Reply:
I have improved this part of the logic.